lib/DateTime.pm

```diff
@@ -2506,11 +2513,8 @@ This class method can be used to construct a new DateTime object from
 an epoch time instead of components. Just as with the C<new()>
 method, it accepts "time_zone", "locale", and "formatter" parameters.
 
-If the epoch value is not an integer, the part after the decimal will
-be converted to nanoseconds. This is done in order to be compatible
-with C<Time::HiRes>. If the floating portion extends past 9 decimal
-places, it will be truncated to nine, so that 1.1234567891 will become
-1 second and 123,456,789 nanoseconds.
+If the epoch value is a floating-point value, it will be rounded to
+nearest microsecond.
```

Comment:
Perl is typically compiled with NV as a double. A double with a significand precision of 53 bits can only represent a nanosecond epoch without loss of precision if the duration from zero epoch is less than ≈ ±104 days. With microseconds the duration is ±104000 days, which is ~ ±285 years.

The code is rounded to nearest **microsecond**, not millisecond!